plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorTest.java

```diff
@@ -0,0 +1,215 @@
+                        .put("hive.s3.path-style-access", "true")
+                        .put("hive.s3.streaming.part-size", "5MB")
+                        .build());
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_RENAME_SCHEMA:
+                return false;
+
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected String createSchemaSql(String schemaName)
+    {
+        return "CREATE SCHEMA IF NOT EXISTS " + schemaName + " WITH (location = 's3://" + bucketName + "/" + schemaName + "')";
+    }
+
+    @Override
+    protected Long calculateFileSystemFileSize(String filePath)
+    {
+        return this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, filePath.replace("s3://" + bucketName + "/", ""))
+                .getObjectMetadata()
+                .getContentLength();
+    }
+
+    @Override
+    protected BufferedInputStream createDataFileStreamForFileSystem(String file)
+    {
+        return new BufferedInputStream(this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, file.replace("s3://" + bucketName + "/", "")).getObjectContent());
+    }
+
+    @Override
+    protected List<String> getAllDataFilesFromTableDirectory(String tableName)
+    {
+        String tableLocation = getSession().getSchema().orElseThrow() + "/" + tableName + "/data";
+        return this.hiveMinioDataLake.getS3Client().listObjects(bucketName, tableLocation)
+                .getObjectSummaries()
+                .stream()
+                .map(object -> "s3://" + bucketName + "/" + object.getKey())
+                .filter(object -> !object.endsWith(".crc"))
+                .collect(toImmutableList());
+    }
+
+    @Test
+    public void testRepartitionDataOnCtasPartitionWithSameColumnMultipleTimes()
+    {
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), 300, true, "'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'", 99);
+    }
+
+    @Test
+    public void testRepartitionDataOnInsertPartitionWithSameColumnMultipleTimes()
+    {
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), 300, false, "'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'", 99);
+    }
+
+    /**
+     * DataProvider overload to exclude single test cases 'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)',
+     * from generic tests:
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnCtas(Session, String, int)}
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnInsert(Session, String, int)}
+     * <p>
+     * They cause ReadTimeout on Thrift when used with dockerized MinIO Data lake. Which leads to TableAlreadyExists.
```

Comment:
> They cause ReadTimeout on Thrift when used with dockerized MinIO Data lake. Which leads to TableAlreadyExists.

Here would be welcomed a better message describing the HMS related issue.

Reply:
Ok reworded. Let me know if this is ok.